src/components/layout/nav/Navbar.jsx

```diff
@@ -1,7 +1,121 @@
-import React from "react";
+import React, { useReducer } from "react";
+import Container from "react-bootstrap/Container";
+import Col from "react-bootstrap/Col";
+import Row from "react-bootstrap/Row";
+import { NavLink } from "react-router-dom";
+import SignUpBox from "./signup/SignUp";
+import SearchBox from "./search/Search";
+import "../../home/HomePage.scss";
+
+function navbarIconsReducer(state, action) {
+  switch (action.type) {
+    case "IS_SIGN_OPENED":
+      return {
+        ...state,
+        isSignBoxOpen: !state.isSignBoxOpen,
+        isSearchBoxOpen: false,
+      };
+
+    case "IS_SEARCH_OPENED":
+      return {
+        ...state,
+        isSearchBoxOpen: !state.isSearchBoxOpen,
+        isSignBoxOpen: false,
+      };
+
+    case "IS_HAMBURGER_OPENED":
+      return {
+        isHamburgerOpen: !state.isHamburgerOpen,
+        isSignBoxOpen: false,
+        isSearchBoxOpen: false,
+      };
+
+    default:
+      return state;
+  }
+}
 
 const Navbar = () => {
-  return <div></div>;
+  const [
+    { isSignBoxOpen, isSearchBoxOpen, isHamburgerOpen, isAdminLoggedIn },
+    dispatch,
+  ] = useReducer(navbarIconsReducer, {
+    isSignBoxOpen: false,
+    isSearchBoxOpen: false,
+    isHamburgerOpen: false,
+    isAdminLoggedIn: false,
```

Comment:
i would guess you want isadminloggedin in redux, it's not really the job of the navbar to determine this